src/types/cql-patient.interfaces.ts

```diff
@@ -1,11 +1,25 @@
-import { Record } from '../cql-patient';
-
+/*
+ * Iterator for the patients provided to the execution engine
+ */
 export interface DataProvider {
-  patients: any[];
   currentPatient(): any;
   nextPatient(): any;
 }
 
+/*
+ * Structure of a Record
+ */
 export interface RecordObject {
-  [recordType: string]: Record[];
+  get(field: any): any;
+  getId(): any;
+  getCode(field: any): any;
+  getDate(field: any): any;
+  getDateOrInterval(field: any): any;
```

Comment:
Hey, so... I think we also need `_is` (used in `runtime/context.ts`) and `_typeHierarchy` (used in `elm/type.ts`) as _optional_ functions (is that a thing you can do in TS)?  Not all implementations will do them (particularly legacy ones), but they're needed to support a few of the operators (like `Is` and `As`).

Reply:
My apologies!!

![image](https://user-images.githubusercontent.com/16297930/154539758-0dd54180-216b-49d1-88f6-ce2014fed90d.png)

Reply:
Added both of these and `_typeHierarchy` structures in [c139c09](https://github.com/cqframework/cql-execution/pull/260/commits/c139c09f7e72b4a6650dfe056132fefd342f24fa)